vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java

```diff
@@ -1,161 +1,174 @@
 
         @Override
-        protected BigDecimal generateModelValue() {
-            try {
-                Function<TextField, Integer> toNumber = t ->
-                   !t.isEmpty() ? Integer.valueOf(t.getValue()) : 0;
-                int i = toNumber.apply(integerPart);
-                int f = toNumber.apply(fractionalPart);
-                return new BigDecimal(i + "." + f);
-            } catch (NumberFormatException e) {
-                return BigDecimal.ZERO;
-            }
+        protected String generateModelValue() {
+            return countryCode.getValue() + " " + areaCode.getValue() + " "
+                    + subscriberCode.getValue();
         }
 
         @Override
-        protected void setPresentationValue(BigDecimal newPresentationValue) {
+        protected void setPresentationValue(String newPresentationValue) {
             if (newPresentationValue == null) {
-                integerPart.setValue(null);
-                fractionalPart.setValue(null);
-            } else {
-                String[] parts = newPresentationValue.toPlainString()
-                    .split("\\.");
-                integerPart.setValue(parts[0]);
-                fractionalPart.setValue(parts.length > 1 ? parts[1] : null);
+                areaCode.setValue(null);
+                subscriberCode.setValue(null);
             }
         }
-
     }
     // end-source-example
 
     // begin-source-example
-    // source-example-heading: Single element wrapping
-    public static class SingleElementWrapping extends CustomField<String> {
-        private final TextField wrappedField = new TextField();
-
-        SingleElementWrapping() {
-            setLabel("Name");
-            add(wrappedField);
-        }
+    // source-example-heading: Value change event
+    public static class ValueChangeEvent extends VerticalLayout {
+        private final Div display = new Div();
 
-        @Override
-        protected String generateModelValue() {
-            return wrappedField.getValue();
+        ValueChangeEvent() {
+            // The code is reused from the previous example.
+            PhoneNumberField phoneNumberField = new PhoneNumberField();
+            phoneNumberField.addValueChangeListener(
+                    e -> display.setText("Value: " + e.getValue()));
+            add(phoneNumberField, display);
         }
+    }
+    // end-source-example
 
-        @Override
-        protected void setPresentationValue(String newPresentationValue) {
-            wrappedField.setValue(newPresentationValue);
+    // begin-source-example
+    // source-example-heading: Required
+    public static class MainForRequired extends VerticalLayout {
+        private final PriceField priceField = new PriceField();
+
+        MainForRequired() {
+            Binder<Product> binder = new Binder<>();
+            Product product = new Product();
+            binder.forField(priceField)
+                    .asRequired("Please fill the price amount")
```

Comment:
Probably some other validation constraints should be added to the field to show the actual validation apart from the `required` field.

Reply:
Based on 1:1 discussion this probably should be changed for all of the components demos at once. Not in scope of this PR